File: WeexBox/Module/Util/UtilModule.m
//
//  UtilModule.m
//  WeexBox
//
//  Created by Mario on 2018/8/18.
//  Copyright © 2018年 Ayg. All rights reserved.
//

#import "UtilModule.h"

#pragma clang diagnostic ignored "-Wundeclared-selector"
@implementation UtilModuleOC

WX_EXPORT_METHOD_SYNC(@selector(getStatusBarHeight))
WX_EXPORT_METHOD_SYNC(@selector(getScreenWidth))
WX_EXPORT_METHOD_SYNC(@selector(getScreenHeight))
WX_EXPORT_METHOD_SYNC(@selector(getWeexWidth))
WX_EXPORT_METHOD_SYNC(@selector(getWeexHeight))

@end
#pragma clang diagnostic pop

File: WeexBox/Module/Util/UtilModule.swift
//
//  UtilModule.swift
//  WeexBox
//
//  Created by Mario on 2018/8/18.
//  Copyright © 2018年 Ayg. All rights reserved.
//

import Foundation

class UtilModule: EventModuleOC {
    
    // 获取状态栏高度
    @objc func getStatusBarHeight() -> Int {
        return Int(getVC().statusBarHeight / weexInstance.pixelScaleFactor)
    }
    
    // 获取屏幕宽度
    @objc func getScreenWidth() -> Int {
        return Int(UIScreen.main.bounds.size.width / weexInstance.pixelScaleFactor)
    }
    
    // 获取屏幕高度
    @objc func getScreenHeight() -> Int {
        return Int(UIScreen.main.bounds.size.height / weexInstance.pixelScaleFactor)
    }
    
    // 获取weex宽度
    @objc func getWeexWidth() -> Int {
        return Int(weexInstance.frame.size.width / weexInstance.pixelScaleFactor)
    }
    
    // 获取weex高度
    @objc func getWeexHeight() -> Int {
        return Int(weexInstance.frame.size.height / weexInstance.pixelScaleFactor)
    }
    
}
